Quickstart baremetal notes

kegilbert opened the pull request:
Add Bare Metal build profile instructions to the quick start. Additionally remove hardcoded expected serial output. The serial output is not updated each release and is liable to change. As such I've removed until we have a method to automatically update the output each patch release.

kegilbert committed:
Add bare-metal build instructions to quick start guide

kegilbert committed:
Remove hardcoded expected serial output.

The serial output is not updated each release and is liable to change. As such I've removed until we have a method to automatically update the output each patch release.

Additionally added a note on the expected output in bare metal builds.

kegilbert committed:
Disable JSON snippet from running in CI

AnotherButler commented:
Does this one also require [ARMmbed/mbed-os#9800](https://github.com/ARMmbed/mbed-os/pull/9800)?

Commit:
Edit cli_code.md

Edit bare metal note for note formatting and consistent capitalization across docs.

Commit:
Delete comma in example_walkthrough.md

Delete unnecessary Oxford comma, and fix note formatting.

Commit:
Edit quick-start-compiler.md

Edit bare metal addition, mostly for note formatting and consistent capitalization across docs.

AnotherButler reviewed (approved):
Thanks for the PR 👍

kegilbert commented:
@AnotherButler Good point, yes it does. The `bare-metal` could be changed to:
```JSON
{
    "requires": ["platform", "drivers"]
}
```
for the same effect, but would probably be better to wait for the other PR to merge.
